docs/p2p.rst

```diff
@@ -4,290 +4,83 @@ P2P Communication
-Thus, a cloud of *Agents* interconnected is create.
+Thus, a cloud of interconnected *Agents* is created.
 
 .. image:: images/p2p_overview.svg
 
-How to use P2P
---------------
-
 Some consideration shall be taken into account in order to use the P2P communication:
 
-#. Only the `create_<entity>_by_ref()` functions shall be used.
+#. Only the :code:`create_<entity>_by_ref()` functions shall be used.
 #. The Topic's reference represents the name of the Topic.
-#. The DataWriter's and DataReader's references shall match the Topic's reference.
-#. Publishers and Subscriber have no roles.
-#. Agents shall use CedMiddleware and Discovery server.
+#. The DataWriter's and DataReader's references need to match the Topic's reference.
+#. Publishers and Subscribers have no role.
+#. Agents use the CedMiddleware and the Discovery mechanism.
+
+Publish/Subscribe P2P Example
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This kind of behaviour can be probed by putting in communication a Publisher P2P application with
+a Subscriber P2P application.
 
-Publish/Subscribe Example
-^^^^^^^^^^^^^^^^^^^^^^^^^
+Agent with CedMiddleware
+------------------------
 
-The following code shows a Publisher and Subscribe P2P application which take into account the aforementioned considerations.
-These *Client* application can be tested launching the following *eProsima Micro XRCE-DDS Agent* application:
+First of all, install the *Agent* as explained in the :ref:`install_agent` section.
+On Linux, this would be: ::
+
+    $ git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git
+    $ cd Micro-XRCE-DDS-Agent
+    $ mkdir build && cd build
+    $ cmake ..
+    $ make
+    $ sudo make install
+
+After having installed the *Agent* system-wide, you can launch it with the :code:`ced` option activated: ::
 
     $ ./MicroXRCEAgent udp -p <port> -m ced -d
 
-.. code-block:: C
+Client P2P publisher app
+------------------------
+
+Let's now install the *Client* locally, and with the :code:`-DUCLIENT_BUILD_EXAMPLES=ON` flag enabled, so as
+to activate the compilation of the examples. On Linux, this implies running the following: ::
+
+    $ git clone https://github.com/eProsima/Micro-XRCE-DDS-Client.git
+    $ cd Micro-XRCE-DDS-Client
+    $ mkdir build && cd build
+    $ cmake .. -DUCLIENT_BUILD_EXAMPLES=ON
+    $ make
+
+At this point, it's possible to launch the :code:`PublishHelloWorldClientP2P` executable
+located in the folder :code:`Micro-XRCE-DDS-Client/build/examples/PublishHelloWorldP2P`, which'll make
+the *Client* publish in the DDS World the :code:`HelloWorld` topic
+(take a look at the IDL defining this topic in the file
+:code:`Micro-XRCE-DDS-Client/examples/PublishHelloWorldP2P/HelloWorld.idl`). ::
+
+    $ Micro-XRCE-DDS-Client/build/examples/PublishHelloWorld2P2/PublishHelloWorldClientP2P 127.0.0.1 2019
+
+.. TODO: decide which path to give to the user.
```

Comment:
@pablogs9 Let's agree on the path.

Reply:
If we have said `At this point,` I think that we should assume that user is in `build` folder, so the command should be:

```bash
 $ examples/PublishHelloWorld2P2/PublishHelloWorldClientP2P 127.0.0.1 2019
```

Change if you agree

Reply:
agree.